Description: use xdg-open to open things in external applications
Credits: novie

diff --git a/Telegram/SourceFiles/platform/linux/file_utilities_linux.cpp b/Telegram/SourceFiles/platform/linux/file_utilities_linux.cpp
index 0437363c3..d787cfcf9 100644
--- a/Telegram/SourceFiles/platform/linux/file_utilities_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/file_utilities_linux.cpp
@@ -24,6 +24,7 @@ namespace Platform {
 namespace File {
 
 void UnsafeOpenUrl(const QString &url) {
+#if 0
 #ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
 	try {
 		if (Gio::AppInfo::launch_default_for_uri(url.toStdString())) {
@@ -37,6 +38,7 @@ void UnsafeOpenUrl(const QString &url) {
 	if (QDesktopServices::openUrl(url)) {
 		return;
 	}
+#endif
 
 	QProcess::startDetached(qsl("xdg-open"), { url });
 }
@@ -56,6 +58,7 @@ bool UnsafeShowOpenWith(const QString &filepath) {
 }
 
 void UnsafeLaunch(const QString &filepath) {
+#if 0
 #ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
 	try {
 		if (Gio::AppInfo::launch_default_for_uri(
@@ -70,11 +73,14 @@ void UnsafeLaunch(const QString &filepath) {
 		return;
 	}
 #endif // !DESKTOP_APP_DISABLE_DBUS_INTEGRATION
+#endif
 
 	const auto qUrlPath = QUrl::fromLocalFile(filepath);
+#if 0
 	if (QDesktopServices::openUrl(qUrlPath)) {
 		return;
 	}
+#endif
 
 	QProcess::startDetached(qsl("xdg-open"), { qUrlPath.toEncoded() });
 }
